src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java

```diff
@@ -26,7 +41,25 @@ public static String getLoopIdentifier(String variableName, VariablesMap variabl
 
 	public static String getParentGroupName(String variableName, VariablesMap variablesMap) {
 		Variable variable = variablesMap.getVariable(variableName);
-		if ( variable == null || variable.getGroup().isRoot()) {
+		List<String> varsEno = Arrays.asList(Constants.getEnoVariables());
+		if ( variable == null ) {
+			if(varsEno.contains(variableName))
+			{
+				return Constants.ROOT_GROUP_NAME;
+			}
+			if(variableName.startsWith(Constants.FILTER_RESULT_PREFIX)
+					&& variablesMap.hasVariable(variableName.replace(Constants.FILTER_RESULT_PREFIX,""))
+			){
+				return variableName.replace(Constants.FILTER_RESULT_PREFIX,"");
```

Comment:
you return the variable and not the group

Reply:
The name of the function is misleading, renamed it to "getParentName"